src/objects/api/validators.py

```diff
@@ -88,3 +89,35 @@ def validate_data_attrs(value: str):
                 "Operator `%(operator)s` supports only dates and/or numeric values"
             ) % {"operator": operator}
             raise serializers.ValidationError(message, code=code)
+
+
+class GeometryValidator:
+    code = "geometry-not-allowed"
+    message = _("This object type doesn't support geometry")
+
+    def set_context(self, serializer):
+        """
+        This hook is called by the serializer instance,
+        prior to the validation call being made.
+        """
+        self.instance = getattr(serializer, "instance", None)
+
+    def __call__(self, attrs):
+        object_type = attrs.get("object_type") or self.instance.object_type
+        geometry = attrs.get("record", {}).get("geometry")
+
+        if not geometry:
+            return
+
+        client = object_type.service.build_client()
```

Comment:
Hmm, this implementation causes 2 calls to Objecttype API, one for the objecttype and one for the geo. Seems inefficient? I can make a new ticket to optimize this but this should not be released as is.

Reply:
Actually I don't know how to reduce the number of calls, because `json_schema` validation requests Objecttype.version and `geometry` validation requests Objecttype itself. 
The only solution is to place `hasGeometry` into Objecttype.version but I don't think it's a good decision